javalib/src/main/scala/java/nio/CharBuffer.scala

```diff
@@ -40,7 +40,7 @@ abstract class CharBuffer private[nio] (
     if (n == 0) -1
     else if (_array != null) {
       // even if read-only
-      target.put(_array, _arrayOffset, n)
+      GenBuffer(target).generic_put(_array, _arrayOffset, n)
```

Comment:
Isn't that a spurious change? In other files you have not changed used methods

Reply:
It's necessary, other files always use the GenBuffer abstraction to avoid changing public api, so there was no need to change anything there. target.put() takes an Array, while GenArray().generic_put takes a GenArray, which is what we have now.